keel-web/src/main/kotlin/com/netflix/spinnaker/keel/apidocs/AbstractSchemaCustomizer.kt

```diff
@@ -20,30 +22,63 @@ abstract class AbstractSchemaCustomizer : BaseModelConverter() {
   ): Schema<*>? =
     super.resolve(annotatedType, context, chain)
       ?.also {
-        val schema = if (annotatedType.isResolveAsRef) {
-          val referencedSchema = context.definedModels[annotatedType.rawClass.simpleName]
-          if (referencedSchema == null) {
-            log.warn("Referenced schema ${annotatedType.rawClass.simpleName} not found")
-            null
-          } else {
-            referencedSchema
+        if (supports(annotatedType.rawClass)) {
+          val schema = resolveIfReference(annotatedType, context, it)
+          if (schema != null) {
+            customize(schema, annotatedType.rawClass, context)
           }
-        } else {
-          it
         }
+      }
 
-        if (schema != null) {
-          customize(schema, annotatedType.rawClass)
-        }
+  private fun resolveIfReference(
+    annotatedType: AnnotatedType,
+    context: ModelConverterContext,
+    schema: Schema<*>
+  ): Schema<*>? =
+    if (annotatedType.isResolveAsRef) {
+      val referencedSchema = context.definedModels[annotatedType.rawClass.simpleName]
+      if (referencedSchema == null) {
+        log.warn("Referenced schema ${annotatedType.rawClass.simpleName} not found")
+        null
+      } else {
+        referencedSchema
       }
+    } else {
+      schema
+    }
+
+  /**
+   * Override this to only process certain classes. By default [customize] will run for all types.
+   */
+  open fun supports(type: Class<*>) = true
 
   /**
    * Provides a hook to customize [schema].
    *
    * @param type the model class the schema represents.
    * @param schema the current schema as generated so far.
    */
-  abstract fun customize(schema: Schema<*>, type: Class<*>)
+  abstract fun customize(schema: Schema<*>, type: Class<*>, context: ModelConverterContext)
+
+  protected inline fun <reified T> Class<*>.isSubclassOf() = T::class.java.isAssignableFrom(this)
+
+  /**
+   * Converts a Kotlin property to the name that would be used in the API schema.
+   */
+  private fun KProperty<*>.toSchemaPropertyName() =
+    if (returnType.javaType == Boolean::class.java) {
+      name.substring(2).decapitalize()
```

Comment:
Maybe add a comment explaining what this is doing? I don't understand the substring part.